python/stempy/image/__init__.py

```diff
@@ -653,3 +653,116 @@ def _electron_counted_metadata_to_dict(metadata):
         ret[name] = getattr(metadata, name)
 
     return ret
+
+def virtual_darkfield(array, centers_x, centers_y, radii, plot=False):
+    """Calculate a virtual dark field image from a set of round virtual apertures in diffraction space.
+    Each aperture is defined by a center and radius and the final image is the sum of all of them.
+    
+    :param array: The SparseArray
+    :type array: SparseArray
+    
+    :param centers_x: The center of each round aperture as the row locations
+    :type centers_x: number or iterable
+    
+    :param centers_y: The center of each round aperture as the column locations
+    :type centers_y: number or iterable
+    
+    :param radii: The radius of each aperture.
+    :type radii: number or iterable
+    
+    :param plot: If set to True then the apertures are plotted as circles using plot_virtual_darkfield
+    :rype plot: bool
+    
+    :rtype: np.ndarray
+    
+    :example:
+    >>> sp = stempy.io.load_electron_counts('file.h5')
+    >>> dp2 = stempy.image.virtual_darkfield(sp, (288, 260), (288, 160), (10, 10)) # 2 apertures
+    >>> dp1 = stempy.image.virtual_darkfield(sp, 260, 160, 10) # 1 aperture
+    
+    """
+    
+    # Change to iterable if single value
+    if isinstance(centers_x, (int, float)):
+         centers_x = (centers_x,)
+    if isinstance(centers_y, (int, float)):
+         centers_y = (centers_y,)
+    if isinstance(radii, (int, float)):
+         radii = (radii,)
```

Comment:
If there's any chance the single values could be a numpy type (like `np.int32`), then this check might not work. But we could also just wait and see if we run into that issue.

Reply:
someone might use an `numpy.ndarray`. Im not sure the best way to test for all of these things. Lets see how it goes as you suggest.